ROADMAP.md

```diff
@@ -1,62 +1,191 @@
+
+#### Mid Q4 (November)
+- [A.2 Interop tests for all existing/developing features](#2-interop-tests-for-all-existingdeveloping-features)
+- [A.3 Rock solid libp2p releases](#3-rock-solid-libp2p-releases)
+- [C.1 Benchmarking using nix-builders](#1-benchmarking-using-nix-builders)
+
+#### End of Q4 (December)
+- [A.4 Canonical Interop Tests & Dashboard](#4-canonical-interop-tests--dashboard)
+
+### 2023
+
+#### Early Q1 (January)
+
+- add here
+
+#### Mid Q1 (February)
+
+- add here
+
+#### End of Q1 (March)
+- [C.2 Benchmarking using remote runners](#2-benchmarking-using-remote-runners)
+
+### Up Next
+
+## 📖 Appendix
+
+**Projects are listed in descending priority.**
+
+### A. [Multi-dimensional Testing/Interop Visibility](https://github.com/libp2p/test-plans/issues/53)
+**Why:** libp2p supports a variety of protocols (multiple transports, muxers, & security protocols) across implementations in different languages. Until we actually test them together, we can't guarantee implementation interoperability. We need to ensure and demonstrate that: interoperable features work with each other as expected and we don't introduce degradations that break interoperability in new releases.
+
+**Goal:** libp2p implementers run tests across permutations of libp2p implementations, versions, and supported protocols. Implementers and users can reference a single website with a dashboard to view the Pass/Fail/Implemented/Not Implemented results of multi-dimensional tests.
+
+**How:**
+#### 1. [User configured interop tests & dashboard](https://github.com/libp2p/test-plans/issues/55)
+Enable test-plans maintainers to define a configuration (of libp2p impls, versions, protocols, expected RTT), run Testground tests per configuration, and retrieve resulting data in a standard format.
+The test case results can be formatted as a "dashboard" (simple Markdown table of Pass/Fail results.)
+
+#### 2. Interop test plans for all existing/developing features
+<!-- TODO: Create issue -->
+Using tooling from A.1, all features of go-libp2p and rust-libp2p that should be interoperable are tested (i.e. transports (TCP, QUIC, WebRTC, WebTransport), multiplexers (mplex, yamux), secure channels (TLS, Noise), etc.) across versions.
+
+Features currently in development across implementations (like WebRTC in go-libp2p and rust-libp2, or QUIC & TLS in rust-libp2p) are not merged without at least manually running these test suites.
+
+Test suites are run in libp2p/test-plans CI.
+
+**Note:** node.js-libp2p is not listed here since it does not have support in Testground yet.
+
+#### 3. Rock solid libp2p releases
+<!-- TODO: Create issue -->
+The full `libp2p/test-plans` test suite is run before releasing a version of go-libp2p,  rust-libp2p, and js-libp2p.
+A GitHub workflow is added so that these suites run for any PR against master and run against the `master` branch on a nightly basis (updating the status check.)
+
+#### 4. Canonical interop tests & dashboard
```

Comment:
I would vote to do this sooner - maybe as #2.  It gives a very clear signal to the community of what we're aiming for and how far we have to go.  It also becomes a clear way for us to show progress each month.  I'd love at monthly all hands for us to show the month over month change of this.

Reply:
Isn’t #2 a prereq for this issue?

Reply:
@MarcoPolo : I don't think it's a prerequ.  I'm viewing this as analogous to other behaviours in software develop;meant like putting in the metric before you make the performance improvement or writing the test before the implementation.  The dashboard makes clear what we're aiming for.  It's our baseline.  To start with it will be red/empty for all/most the combinations.  Then as we do A.2 "Interop test plans for all existing/developing features" we'll start showing the matrix progressively filling more-and-more each week.  

I think this came up in a verbal conversation with @mxinden as well (which is making me think it would be good to use sync time in Lisbon to be on the same page here), but I'm not suggesting to prioritize the dashboard so we can claim interoperability or pad ourselves on the back.  I want so we have a clear target that we can show steady progress against.  

Does that make sense?

Reply:
> To start with it will be red/empty for all/most the combinations.

IIUC, I think this should still be possible A.1's simple dashboard, but it'll be a markdown output and require a manual effort to: trigger to run the interop test plans and generate/update the resulting dashboard.
 
the comprehensiveness of the simple dashboard would be dependent on the permutations specified in the resource/composition file. I think we can specify all permutations as soon as the tooling is ready. It should also be possible to list incompatible incompatible/unsupported.

>It also becomes a clear way for us to show progress each month

Having said that I think A.1 can meet this goal.

A.3 is more about generating a nicer output that can be shared on a website, one which is updated from CI without manual intervention.

Reply:
I'm good to be scrappy here.  At the minimum I want to make sure we can produce/show the interoperability dashboard with its dimensionality monthly during our all-hands (and ideally highlight the deltas).  Even better (but not required) would be if this was a primary "metric" reported by libp2p in the weekly sitrep.  I'm good with however we get there.  If it's a major manual effort then it won't happen, so I'm good to invest in automation early here.  Manually starting a github action on a monthly basis to generate the data seems good enough to me to start.  Thanks all.